src/ops/fixit.rs

```diff
@@ -85,37 +94,48 @@ fn exec(args: FixitArgs) -> CargoResult<()> {
     Ok(())
 }
 
+fn check(args: &FixitArgs) -> CargoResult<(impl Iterator<Item = CheckMessage>, Option<i32>)> {
+    let cmd = if args.clippy { "clippy" } else { "check" };
+    let command = std::process::Command::new(env!("CARGO"))
+        .args([cmd, "--message-format", "json"])
+        .args(args.check_flags.to_flags())
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let buf = BufReader::new(Cursor::new(command.stdout));
+
+    Ok((
+        buf.lines()
+            .map_while(|l| l.ok())
+            .filter_map(|l| serde_json::from_str(&l).ok()),
+        command.status.code(),
+    ))
+}
+
 #[tracing::instrument(skip_all)]
-fn run_rustfix(
+#[allow(clippy::type_complexity)]
+fn collect_errors(
     args: &FixitArgs,
-    files: &mut IndexMap<String, File>,
-    current_target: &mut Option<(Target, String)>,
-    seen: &HashSet<(Target, String)>,
-) -> CargoResult<(IndexSet<String>, bool)> {
+    current_target: &mut Option<BuildUnit>,
+    seen: &HashSet<BuildUnit>,
+) -> CargoResult<(
+    IndexSet<String>,
+    IndexMap<String, IndexSet<(Suggestion, Option<String>)>>,
+)> {
     let only = HashSet::new();
     let mut file_map = IndexMap::new();
 
     let mut errors = IndexSet::new();
 
-    let cmd = if args.clippy { "clippy" } else { "check" };
-    let mut command = std::process::Command::new(env!("CARGO"))
-        .args([cmd, "--message-format", "json"])
-        .args(args.check_flags.to_flags())
-        .stderr(Stdio::piped())
-        .stdout(Stdio::piped())
-        .spawn()?;
+    let (messages, _exit_code) = check(args)?;
```

Comment:
My assumption was that the caller would call `check` and the results would be passed to this function